icgc-argo/platform-ui
# 2301 - update apollo & graphql

# Description of changes

- upgrade to latest versions of apollo/client and graphql
- migrate various libraries to apollo/client 3
- remove graphql test step in Jenkins build
- add graphql codegen to check types and schema in development
- refactor some files due to Apollo throwing errors
  - variables used in some GQL commands were declared after the command
  - one instance of modifying the response object
- move `*.gql` files to `gql/*_[QUERY/MUTATION/FRAGMENT].ts` and update naming to follow GQL conventions
- update README
- update cache IDs due to API changes in Apollo 3

## Type of Change


## Checklist before requesting review

- Design (select one):
    - component sizes, spacing, and styles
    - font size, weight, colour
    - spelling has been double checked
  - [ ] No design provided, screenshot of changes included in PR
  - [x] No changes to UI
- [ ] Feature is minimally responsive
- [x] Manual testing of UI feature
- [x] Add copyrights to new files
- [x] Connected ticket to PR

https://app.zenhub.com/workspaces/icgc-argo-platform-dk-production-board-5e542d38415f5034e9fed89d/issues/icgc-argo/platform-ui/2301
